File: Sliding Window/Longest Subarray of 1's After Deleting One Element/solution.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let (mut result, mut prev, mut cur) = (0,0,0);
        for &num in nums.iter() {
            if num == 0 {
                prev = cur;
                cur = 0;
            } else {
                cur += 1;
                result = std::cmp::max(result, prev + cur);
            }
        }
        return std::cmp::min(result, nums.len() as i32 - 1)
    }
}

File: Sliding Window/Longest Subarray of 1's After Deleting One Element/solution.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int result = 0;
        for(int i = 0, j = 0, k = 0; i < nums.size(); ++i) {
            if(nums[i] == 0) ++k;
            while(k > 1) {
                k -= nums[j++] == 0;
            }
            result = max(result, i - j);
        }
        return result;
    }
};
